lace/emulator/nn_emulator.py

```diff
@@ -262,26 +268,20 @@ def _obtain_sim_params(self):
             Nk (int): Number of k values.
             k_Mpc_train (tensor): k values in the k training range
         """
-
-        # AFR: not sure exactly what we are doing here. 
-        # It looks like some code to set self.k_Mpc and self.Nz
-
-        # AFR: this could be dangerous if not all sims have the same Nz (Nyx)
-        sim_0 = copy.deepcopy(self.archive)
-        sim_0_training_data = sim_0.get_training_data(emu_params=self.emu_params)
-        sim_0_training_data = [d for d in sim_0_training_data if d["sim_label"] == "mpg_0"]
-        sim_zs = [data["z"] for data in sim_0_training_data]
-        Nz = len(sim_zs)
-        k_Mpc = sim_0.data[0]["k_Mpc"]
-        self.k_Mpc = k_Mpc
-
-        k_mask = (k_Mpc < self.kmax_Mpc) & (k_Mpc > 0)
-        k_Mpc_train = k_Mpc[k_mask]
+        
+        k_mask = [
+            (self.training_data[i]["k_Mpc"] < self.kmax_Mpc) & (self.training_data[i]["k_Mpc"] > 0)
+            for i in range(len(self.training_data))
+        ]
+        
+        self.k_mask= k_mask
+        
+        k_Mpc_train = self.training_data[0]["k_Mpc"][k_mask[0]]
```

Comment:
I see that you are using the first sim in the training data. Are we sure that the k-range is the same for all the sims?

Reply:
Some sims have an additional k bin, but to k <4 all k are the same